sdk/iot/Azure.Iot.Hub.Service/samples/IotHubClientSamples/ModuleIdentityLifecycleSamples.cs

```diff
@@ -30,22 +32,19 @@ public async Task RunSampleAsync()
             ModuleIdentity moduleIdentity = await CreateModuleIdentityAsync(deviceIdentity, moduleId);
 
             // List all modules within the device.
-            await ListAllModulesAsync();
+            await ListAllModulesAsync(deviceIdentity);
 
             // Get the Module Identity.
-            await GetModuleIdentityAsync();
+            moduleIdentity = await GetModuleIdentityAsync(deviceIdentity, moduleIdentity.ModuleId);
 
             // Update Module Identity.
-            await UpdateModuleIdentityAsync();
+            moduleIdentity = await UpdateModuleIdentityAsync(moduleIdentity);
```

Comment:
this call is confusing as well, we are doing a "get" -> "update" using the same instance, while it should be "get" -> "modify" -> "update". I understand that the "modify" part of it would be happening inside `UpdateModuleIdentityAsync`, but for someone taking a look at this for the first time, this is definitely confusing.

Is there a different way to do this? Perhaps not encapsulate the logic within a method call, and have it here directly?